cou/utils/juju_utils.py

```diff
@@ -434,7 +433,22 @@ async def wait_for_idle(self, timeout: int, apps: Optional[list[str]] = None) ->
         :param apps: Applications to wait, defaults to None
         :type apps: Optional[list[str]], optional
         """
-        model = await self._get_model()
-        await model.wait_for_idle(
-            apps=apps, timeout=timeout, idle_period=DEFAULT_MODEL_IDLE_PERIOD
-        )
+
+        @retry(timeout=timeout)
+        @wraps(self.wait_for_idle)
+        async def _wait_for_idle() -> None:
+            # NOTE(rgildein): Defining wrapper so we can use retry with proper timeout
+            model = await self._get_model()
+            try:
+                await model.wait_for_idle(
+                    apps=apps, timeout=timeout, idle_period=DEFAULT_MODEL_IDLE_PERIOD
+                )
+            except asyncio.exceptions.TimeoutError as error:
+                if apps is None:
+                    msg = f"Model {self.name} did not reach its idle state in {timeout}s"
+                else:
+                    msg = f"Apps {','.join(apps)} did not reach their idle state in {timeout}s"
```

Comment:
The error message would be incorrect if only a subset of the apps have not reached idle state. The ideal approach would be providing the apps that are actually not at idle state. If that's too complicated, we should at very least change the error message to the something like:
```suggestion
                    msg = f"Not all apps {','.join(apps)} reached their idle state in {timeout}s"
```